lib/puppet/type/zfs_share.rb

```diff
@@ -0,0 +1,141 @@
+Puppet::Type.newtype(:zfs_share) do
+  @doc = "Manage zfs shares"
+
+  require 'puppet/parameter/boolean'
+
+  ensurable
+
+  newparam(:zfs_name) do
+    desc "Name of zfs file system"
+    isnamevar
+
+    validate do |value|
+      unless value =~ /(?!\/|\d)^\w.*/
+        raise ArgumentError , "%s is not a valid zfs path."
+      end
+    end
+  end
+
+  newparam(:zpool) do
```

Comment:
To allow a user to set a pool as a default value for example. However, if a user wants to they can do that without this parameter so I've removed it.